### ISSUE TITLE
VPN previous state not restored after interrupting the new tutorials

### ISSUE THREAD

**Issue by data-sync-user** [mozilla-mobile/mozilla-vpn-client#5356]
**Affected versions:**

* 2.13.0 (2.202212301144);

**Tested Platforms:**

* Windows 11;

* Android 10;

**Prerequisites:**

* Have Mozilla VPN installed and opened;

* Have a Fx account with an active subscription;

* Be signed in;

* Have the VPN in ON state;

**Steps to reproduce:**

1. Navigate to Settings - VPN tips and tricks screen;
1. Start the “How to exclude apps from VPN protection” tutorial;
1. Press the “Continue” button from the “Continue tutorial” modal;
1. Press the toggle button, as indicated by the tutorial;
1. Press the “x” button from the “Select your account settings” banner;
1. Interrupt the tutorial by pressing the “Come back later” button from the modal;
1. Observe the behavior;

**Expected result:**

* The VPN is in ON state, as the previous settings before starting the tutorial should be restored; 

**Actual result:**

* The VPN is in OFF state;

**Notes:**

* See the attached video and logs;

┆Issue is synchronized with this [Jira Bug](https://mozilla-hub.atlassian.net/browse/VPN-3593)
┆Reporter: Bianca Hidecuti

**Comment by data-sync-user** [comment]
➤ Bianca Hidecuti commented:

This is also reproducible for the multi-hop tutorial as well.

**Comment by data-sync-user** [comment]
➤ Bianca Hidecuti commented:

Verified this as fixed on Mozilla VPN 2.13.0 (2.202301120432), using Windows 11 and Android 10, Limux - Jammy 22.04 (VM) and on Mozilla VPN 2.13.0 (2.202301120451), using MacOS 12.6.1 - for the multi-hop tutorial.

VPN previous state is now restored after interrupting the “How to exclude apps from VPN protection” / “Using multi-hop VPN” tutorials, while starting them in ON state. 

Attaching postfix video for the Windows / Android and MacOS.

Matt Lichtenstein, however, for Linux there is a difference between Kinetic and Jammy versions → while on Kinetic, the VPN remains OFF after interrupting the new tutorials - see the attached video. Should we reopen this or track it separately?

  * !kinetic.gif|width=601,height=938!

Thank you. 

!3593 app permission.mp4|width=448,height=816!

!3593 multi-hop.mov|width=724,height=1302!

**Comment by data-sync-user** [comment]
➤ Matt Lichtenstein commented:

Bianca Hidecuti nice find, my gut instinct is to track the Kinetic issue separately since the current solution fixes the issue for most platforms, but maybe Santiago Andrigo should advise.

**Comment by data-sync-user** [comment]
➤ Santiago Andrigo commented:

Agreed. I wouldn’t want to block solely on Kinetic. Can we file a distinct bug specific to Kinetic on this?

**Comment by data-sync-user** [comment]
➤ Bianca Hidecuti commented:

Santiago Andrigo / Matt Lichtenstein, while using the latest build from 2.13 qa branch, on Kinetic, the issue is no longer reproducible. If we will encounter the issue again, we will log a separate issue.

Marking this verified as fixed.
